### PR TITLE
Migrate sosreport playbook to role

### PR DESCRIPTION
**Pull request by ra-rau**
- Migrate the sosreport playbook to ansible.commons.roles
- This is partly of osism/ansible-playbooks#224

Signed-off-by: Ramona Rautenberg <rautenberg@osism.tech>

### CODE REVIEW

File: roles/sosreport/meta/main.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+---
+galaxy_info:
+  author: Christian Berendt
```
Comment (berendt) [inline roles/sosreport/meta/main.yml:3]: Please use your own name here.

File: roles/sosreport/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,82 @@
+---
+- name: sosreport - install required packages
```
Comment (berendt) [inline roles/sosreport/tasks/main.yml:2]: ``sosreport -`` as prefix should be removed everywhere

File: roles/sosreport/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  become: true
+  ansible.builtin.package:
+    name: "{{ item }}"
+    state: present
+  loop: "{{ required_packages_sosreport }}"
```
Comment (berendt) [inline roles/sosreport/tasks/main.yml:7]: Use ``sosreport_required_packages`` as name here.

File: roles/sosreport/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    path: "{{ sosreport_tmpdir }}"
+    state: absent
+
+- name: sosreport - create required directories
+  become: true
+  ansible.builtin.file:
+    path: "{{ item }}"
+    state: directory
+    owner: "{{ operator_user }}"
+    group: "{{ operator_group }}"
```
Comment (berendt) [inline roles/sosreport/tasks/main.yml:21]: Add ``operator_user`` + ``operator_group`` to the defaults.

File: roles/sosreport/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+  delegate_to: localhost
+
+- name: sosreport - generate report
+  become: true
+  ansible.builtin.command: "sosreport --batch --tmp-dir {{ sosreport_tmpdir }} -z gzip --name {{ inventory_hostname_short }} -o {{ sosreport_plugins|join(',') }}"
+  changed_when: true
+
+- name: sosreport - rename md5 file
+  ansible.builtin.shell: "mv -f {{ sosreport_tmpdir }}/*.md5 {{ sosreport_tmpdir }}/{{ sosreport_archive_filename }}.md5"
+  changed_when: true
```
Comment (berendt) [inline roles/sosreport/tasks/main.yml:41]: Is it possible to do a move with an Ansible task?

File: roles/sosreport/tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+- name: sosreport - rename archive
+  ansible.builtin.shell: "mv -f {{ sosreport_tmpdir }}/*.tar.gz {{ sosreport_tmpdir }}/{{ sosreport_archive_filename }}"
```
Comment (berendt) [inline roles/sosreport/tasks/main.yml:44]: Is it possible to do a move with an Ansible task?

**Review by berendt — CHANGES_REQUESTED** [review]
Please also add a molecule files + a new GitHub action for this role.

File: roles/sosreport/tasks/main.yml (hunk 5 above)
Comment (ra-rau) [reply roles/sosreport/tasks/main.yml:41]: Found this: https://stackoverflow.com/questions/24162996/how-to-move-rename-a-file-using-an-ansible-task-on-a-remote-system
Up until now there isn´t another way.
But i think i will use the command module.

File: roles/sosreport/tasks/main.yml (hunk 2 above)
Comment (ra-rau) [reply roles/sosreport/tasks/main.yml:2]: Done

File: roles/sosreport/tasks/main.yml (hunk 3 above)
Comment (ra-rau) [reply roles/sosreport/tasks/main.yml:7]: Done

File: roles/sosreport/tasks/main.yml (hunk 4 above)
Comment (ra-rau) [reply roles/sosreport/tasks/main.yml:21]: Done

**Pull request merged**